### PR TITLE
docs(readme): Add TypeFactory usage

### LINKED ISSUE #62: Add `TypeFactory` feature to the readme
One of the features that makes `assertive-ts` different is the `TypeFactory` feature. We should highlight this on the README.md and add some examples in the `Usage` section. Code examples will also make visible the exported `TypeFactories` object which provides some basic factories.

Additionally, we should explain what to do when the expected value is of type `any`. Because of the function overload, we hit a TypeScript limitation, and the first overload matches (because `any` does extend from the first overload's type). The best approach for these cases is to use the `.asType(TypeFactory)` method.

### PR DESCRIPTION
**Pull request by JoseLion**
Resolves #62

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,75 @@ expect(14).toEndWith("4");
 
 For a list of all matchers and extended documentation, please refer to the [API documentation](https://stackbuilders.github.io/assertive-ts/docs/build/).
 
+### Type Factory 🏭
+
+A great features of AssertiveTS is the type safety across the API. But, what should you do if you want to check if the value under test is of an specific type during runtime? The answer is simple, AssertiveTS provides a `.asType(TypeFactory)` method, where the [TypeFactory](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/TypeFactory.html) parameter let's you check for the specific type and let's you cast the assertion to a more specific one. To make things simpler, AssertiveTS provides [TypeFactories]https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/StaticTypeFactories.html) for the basic types:
```
Comment (byrpatrick) [inline README.md]: 💯 It explains the feature nicely!
  
  I found a couple of typos and a missing `(` here:

```suggestion
A great feature of AssertiveTS is the type safety across the API. But, what should you do if you want to check if the value under test is of a specific type during runtime? The answer is simple, AssertiveTS provides a `.asType(TypeFactory)` method, where the [TypeFactory](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/TypeFactory.html) parameter lets you check for the specific type and lets you cast the assertion to a more specific one. To make things simpler, AssertiveTS provides [TypeFactories](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/StaticTypeFactories.html) for the basic types:
```

File: README.md [hunk 2]
```diff
@@ -81,6 +81,75 @@ expect(14).toEndWith("4");
...
+expect(list)
+  .asType(TypeFactories.array(TypeFactories.Number))
+  .toHaveSameMembers([1, 2, 3, 4, 5]);
+```
+
+If the built-in type factories are not enough to assert your specific type, you can always create your own factory. A `TypeFactory<S, A>` is nothing more than an object with 3 properties:
+
+- `Factory: new(actual: S) => A` - The specific assertion constructor to return if the predicate is true. Where `S` is the actual value type, and `A` is the type of the assertion to return (`A` should extend from `Assertion<S>`).
+- `predicate(value: unknown): value is S` - A predicate function that check's if the value is of the expected type.
+- `typeName: string` - The name of the cheked type. Used to make the assertion error message clearer.
```
Comment (byrpatrick) [inline README.md:104]: I found a couple of typos here:

```suggestion
- `predicate(value: unknown): value is S` - A predicate function that checks if the value is of the expected type.
- `typeName: string` - The name of the checked type. Used to make the assertion error message clearer.
```

File: README.md [hunk 3]
```diff
@@ -81,6 +81,75 @@ expect(14).toEndWith("4");
...
+      && "z" in value
+      && Object.values(value).every(v => typeof v === "number");
+  },
+  typeName: "Point3D"
+});
+```
+
+### Handling TypeScript Unions
+
+[Union types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) are a TypeScript concept that is only applicable at type level. During runtime, the value can only be one of the types. For instance, if we say `const foo: number | string = ...`, at runtime `foo` will be either a `number` or a `string`. If you want to use a more specific assertion on a union type, you can use `.asType(..)` to fisrt assert the expected type, and then move forward with more assertions:
```
Comment (byrpatrick) [inline README.md:131]: A typo here:

```suggestion
[Union types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) are a TypeScript concept that is only applicable at type level. During runtime, the value can only be one of the types. For instance, if we say `const foo: number | string = ...`, at runtime `foo` will be either a `number` or a `string`. If you want to use a more specific assertion on a union type, you can use `.asType(..)` to first assert the expected type, and then move forward with more assertions:
```

File: README.md [hunk 4]
```diff
@@ -81,6 +81,75 @@ expect(14).toEndWith("4");
 
 For a list of all matchers and extended documentation, please refer to the [API documentation](https://stackbuilders.github.io/assertive-ts/docs/build/).
 
+### Type Factory 🏭
+
+A great features of AssertiveTS is the type safety across the API. But, what should you do if you want to check if the value under test is of an specific type during runtime? The answer is simple, AssertiveTS provides a `.asType(TypeFactory)` method, where the [TypeFactory](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/TypeFactory.html) parameter let's you check for the specific type and let's you cast the assertion to a more specific one. To make things simpler, AssertiveTS provides [TypeFactories](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/StaticTypeFactories.html) for the basic types:
+
+```ts
+import { expect, TypeFactories } from "@stackbuilders/assertive-ts";
```
Comment (byrpatrick) [inline README.md:89]: @JoseLion I'm not sure if we're able to import the TypeFactories like this.

File: README.md [hunk 5]
```diff
@@ -81,6 +81,75 @@ expect(14).toEndWith("4");
 
 For a list of all matchers and extended documentation, please refer to the [API documentation](https://stackbuilders.github.io/assertive-ts/docs/build/).
 
+### Type Factory 🏭
+
+A great features of AssertiveTS is the type safety across the API. But, what should you do if you want to check if the value under test is of an specific type during runtime? The answer is simple, AssertiveTS provides a `.asType(TypeFactory)` method, where the [TypeFactory](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/TypeFactory.html) parameter let's you check for the specific type and let's you cast the assertion to a more specific one. To make things simpler, AssertiveTS provides [TypeFactories](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/StaticTypeFactories.html) for the basic types:
```
Comment (byrpatrick) [inline README.md:86]: @JoseLion the explanation here is really nice!! 
I found a couple of typos and a missing `(`.
```suggestion
A great feature of AssertiveTS is the type safety across the API. But, what should you do if you want to check if the value under test is of a specific type during runtime? The answer is simple, AssertiveTS provides a `.asType(TypeFactory)` method, where the [TypeFactory](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/TypeFactory.html) parameter lets you check for the specific type and lets you cast the assertion to a more specific one. To make things simpler, AssertiveTS provides [TypeFactories](https://stackbuilders.github.io/assertive-ts/docs/build/interfaces/StaticTypeFactories.html) for the basic types:
```

**Review by byrpatrick — CHANGES_REQUESTED** [review]
@JoseLion is looking good!!

I found a couple of typos. But I'm not approving it because I think the `TypeFactories` is not exported at the same level as `expect`.

File: README.md (hunk 4 above)
Comment (JoseLion) [reply README.md:89]: Great catch @byrpatrick! This is fixed by #67!

**Review by byrpatrick — APPROVED** [review]
LGTM, nice documentation!  🚀 💯

**Pull request merged**